server/src/internalClusterTest/java/org/elasticsearch/indices/DataStreamIT.java

```diff
@@ -433,6 +444,87 @@ public void testCannotDeleteComposableTemplateUsedByDataStream() throws Exceptio
         assertTrue(maybeE.isPresent());
     }
 
+    public void testChangeTimestampFieldInComposableTemplatePriorToRollOver() throws Exception {
+        createIndexTemplate("id1", "logs-foo*", "@timestamp");
```

Comment:
Should we also include tests of any mapping specialization/params? AFAICS, the implementation could keep only the mapping type now and all tests would succeed.